lib/teleterm/daemon/daemon.go

```diff
@@ -187,44 +188,42 @@ func (s *Service) removeGateway(gateway *gateway.Gateway) error {
 		return trace.Wrap(err)
 	}
 
-	// remove closed gateway from list
-	for index := range s.gateways {
-		if s.gateways[index] == gateway {
-			s.gateways = append(s.gateways[:index], s.gateways[index+1:]...)
-			return nil
-		}
-	}
+	delete(s.gateways, gateway.URI.String())
 
-	return trace.NotFound("gateway %v not found in gateway list", gateway.URI.String())
+	return nil
```

Comment:
This looks like an API change of `Service.removeGateway()` but in reality that `trace.NotFound` error was a superfluous check. `removeGateway()` already receives a gateway instance as an argument and all public methods ensure that the gateway you're trying to find is in `s.gateways`.

So if you call `Service.RemoveGateway()` with a nonexistent gateway URI, you're still going to get a not found error.